### ISSUE TITLE
Chrome: tooltip for toolbar icon has extra text

### ISSUE THREAD

**Issue by monzug** [alpheios-project/webextension#179]
tested in build 2.1.0.14 in Chrome.
tooltip says: "Deactivate Apheios Has access to this site"  and 
"Activate Alpheios Has access to this site"

**Comment by kirlat** [comment]
I think it is a Chrome specific tooltip that indicates what permissions a webextension has:
https://chromium.googlesource.com/chromium/src/+/master/chrome/app/generated_resources.grd#3892

**Comment by balmas** [comment]
yes, probably we can't do anything about that. Just curious - what version of Chrome are you running @monzug ? I don't get that in mine (version 69) , but I might be a few updates behind.

**Comment by monzug** [comment]
interesting. my Chrome version is Version 70.0.3538.110 (Official Build) (64-bit)

**Comment by kirlat** [comment]
I do see it on 71.0.3578.80 too.

**Comment by balmas** [comment]
This is probably related to the extension security changes in Chrome 70 described at 
https://blog.chromium.org/2018/10/trustworthy-chrome-extensions-by-default.html and 
https://developer.chrome.com/extensions/runtime_host_permissions

It isn't explicitly mentioned in there but they introduced a set a of changes intended to increase user visibility and control over extension permissions.

@kirlat Can you review the changes and let me know if you see any that we ought to take into account?  Can you also verify that our code minification techniques are in compliance with their updated Content Security Policy (https://developer.chrome.com/webstore/program_policies#content_policies) and recommended minification techniques (https://developers.google.com/speed/docs/insights/MinifyResources)? They will remove any extensions that are not in compliance from the chrome webstore on January 1.

**Comment by kirlat** [comment]
Will check it out

**Comment by kirlat** [comment]
There are three possible settings of user controls for host permissions: "On all sites" (this is a default now and this is the one we've tested our extension with), "On a specific site" (on sites that do not have any explicit permissions it requires user to click on either an icon or a menu to allow access for a site), "When you click the extension" (as previous, but requires user to click for all sites).

If extensions requires a click to be activated on a page, its icon is shown in a grayed circle:
![image](https://user-images.githubusercontent.com/18631055/49953613-ee2c2000-ff17-11e8-8bbe-0b825bf230e5.png)

This is pretty much OK for for all workflows because a majority of them require user to interact with the icon or the menu and that will unblock the extension. There is one exception, though. It is a page reload scenario, where we move to the other page and try to restore a state from a previous page. In this case permissions won't allow state reactivation unless user clicks on the icon. Right now it is a little messed up because background thinks extension is in an active state and changes its state object correspondingly. The extension is, however, deactivated, so clicking on an icon brings extension to a deactivated state instead of activating it. Clicking on the icon again do the activation, but the whole thing does not feel quite right.

In order to handle this we need to check permissions before trying to restore state on a new page. There could be different ways to do that. Unfortunately, that would involve changes of a control flow and more conditional logic. Don't know how critical it is, but ideally it would be preferable to combine this change with some other changes in webextension to avoid a lengthy testing cycle. Decision probably depends on how many people would start to use this feature and how important the state reloading on an inter page navigation is for those who enabled permission controls.

Checking other requirement updates now ...

**Comment by kirlat** [comment]
I think we fully obey rules of Google's Content Policies. We do very minimal minification (can do more, I think) and we do not obfuscate any pieces of code. We do satisfy all other requirements too.

Once we start collecting user data, we should clearly specify what data is collected and how it might be used.

**Comment by balmas** [comment]
super, thanks. Closing this.

**Comment by balmas** [comment]
I have entered a separate issue to track whether or not we want to detect and respond to user overrides of host permissions for the issue @kirlat describes above.
